spec/unit/lib/temporal/workflow/dispatcher_spec.rb

```diff
@@ -43,20 +46,42 @@
       end
 
       it 'stores the target and handler once' do
-        expect(handlers[target]).to be_kind_of(Array)
+        expect(handlers[target]).to be_kind_of(Hash)
         expect(handlers[target].count).to eq 1
       end
 
       it 'associates the event name and handler name with the target' do
-        event = handlers[target].first
+        event = handlers[target][1]
         expect(event.event_name).to eq(event_name)
       end
 
       it 'associates the handler with the target' do
-        event = handlers[target].first
+        event = handlers[target][1]
         expect(event.handler).to eq(block)
       end
     end
+
+    it 'removes a given handler against the target' do
+      block1 = -> { 'handler body' }
+      block2 = -> { 'other handler body' }
+      block3 = -> { 'yet another handler body' }
+
+      handle1 = subject.register_handler(target, 'signaled', &block1)
+      subject.register_handler(target, 'signaled', &block2)
+      subject.register_handler(other_target, 'signaled', &block3)
+
+      expect(subject.send(:handlers)[target][1].event_name).to eq('signaled')
```

Comment:
I'd originally written tests only against the public interface, but other tests were added in #157 that further inspected the private contents. That was merged between when I first started writing this and began upstreaming this PR. So I decided to stick with that convention, and update the tests. Happy to rework this if everyone agrees we should only test the public interface here. This would make the tests less brittle.

The ordering tests are critical, as that's the crux of one of the bugs where the order was not deterministic. I agree the contents of the dispatcher's internal state don't need to be in a particular order, as it's an implementation detail. All that needs to be guaranteed is dispatch order follows registration order. There is a test for this farther down the file already. Do you think that's sufficient? Or are there more cases I could cover there?

Reply:
Ha, yes, and it's particularly annoying to me that I did it in my own PR. :)

Now I'm inspired to go back and fix that because I can't allow myself to be a hypocrite.

As for ordering, I think the later tests cover it sufficiently.

Reply:
I'll leave this for now since at least it has good coverage